### ISSUE TITLE
github_team_member are always "maintainers" on first apply

### ISSUE THREAD

**Issue by jkamenik** [integrations/terraform-provider-github#1046]
### Terraform Version

```
# terraform -v
Terraform v1.1.2
on linux_amd64
+ provider registry.terraform.io/hashicorp/github v4.19.2

Your version of Terraform is out of date! The latest version
is 1.1.4. You can update by downloading from https://www.terraform.io/downloads.html
```

### Affected Resource(s)
- github_team_membership

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
variable "members" {
  type = set(string)
  default = [
    "jkamenik", # me
  ]
}

resource "github_team" "test" {
  name    = "test-group"
  privacy = "closed"
}

resource "github_team_membership" "member" {
  for_each = var.members

  team_id  = github_team.test.id
  username = each.value
  role     = "member" # doesn't matter if this is explicit or not
}
```

### Expected Behavior
That the user are added as "members"

### Actual Behavior
The users are added as "maintainers" the first time.  Then you have to turn a second plan and apply to make them "members".  You can see this by running a plan right after the first apply.

```
terraform apply
...

terraform plan
...
Terraform will perform the following actions:

  # module.test-members.github_team_membership.member["jkamenik"] will be updated in-place
  ~ resource "github_team_membership" "member" {
        id       = "5591725:jkamenik"
      ~ role     = "maintainer" -> "member"
        # (3 unchanged attributes hidden)
    }
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply` - They were made maintainers
2. `terraform plan --out my.plan` - This will show the issues
3. `terraform apply my.plan` - Fixes the issues

**Comment by Satak** [comment]
I can confirm this. We are running on the same issue.

**Comment by jechtom-srovnejto** [comment]
I can confirm that this happens to me as well. 
It seems this happens only with organizational owners. For regular members, this works fine.
The solution for me is to add all organizational owners as maintainers instead of members.

**Comment by bh0520** [comment]
I also have this problem with version = "4.26.1" for both `github_team_membership` and  `github_team_members`.  Members are always added as maintainers even if role is set to "member".

**Comment by ystndr** [comment]
I can confirm this on `integrations/github v5.5.0`. 
Organization owner **cannot** be a team member. 

Running `apply` seemed to make it a "member", but running `plan` after that would show something like this
<img width="401" alt="image" src="https://user-images.githubusercontent.com/31244806/199423558-d1185f6a-2e69-48a3-b0ab-50826c88bac4.png">

**Comment by jcogilvie** [comment]
@jechtom-srovnejto I can confirm that keying off of the user's org-wide github role to decide whether to grant `member` (to non-admins) or `maintainer` (to admins) worked to resolve this perpetual diff in my case.  Thank you!

**Comment by kfcampbell** [comment]
Hmm...if we tried @jcogilvie's workaround at the provider-level, it would cost extra API calls to check the user's status, but we could potentially ensure we requested only a valid membership status. 

However, it wouldn't be super intuitive to me if I requested "member" for an org admin and the call succeeded but the user was made a "maintainer", not a "member". 

My vote is to document this behavior on the provider docs site [here](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/team_members) and advise the mentioned procedure to avoid errors in the API calls. 

Thoughts?
